server/server.js

```diff
@@ -293,6 +294,36 @@ app.get('/logout', (req, res) => {
     }
 });
 
+app.post('/flic_request', header(['button-serial-number','button-battery-level']).exists(), async (req, res) => {
+
+    try {
+        const validationErrors = validationResult(req)
+
+        if(validationErrors.isEmpty()){
+            let button = await db.getButtonWithSerialNumber(req.get('button-serial-number'))
+            if(button === null) {
+                log(`Bad request: Serial Number is not registered. Serial Number is ${req.get('button-serial-number')}`);
+                res.status(400).send();
+            }
+            else {
+                await db.saveButtonBatteryLevel(req.get('button-serial-number'), req.get('button-battery-level'))
+                await handleValidRequest(button, 1)
```

Comment:
we're assuming 1 button press, but this integration will be set up in the Flic app for all 3 button press types - I think we should handle the double click case here

Reply:
there's a problem with this in that we don't actually get the click type in the http request unless we type it in there ourselves

Reply:
hmmm ok

Reply:
see latest push